internal/resolvers/types.go

```diff
@@ -0,0 +1,11 @@
+package resolvers
+
+import "github.com/bluesky-social/indigo/atproto/crypto"
+
+type HabitatHostResolver func(did string) (habitatHost string, err error)
+
+type PublicKeyResolver func(did string) (publicKey crypto.PublicKey, err error)
```

Comment:
thoughts ... I wonder if in the future we should do ... we could probably transform the `ExternalHabitatUser` into this
```
type Identity interface {
     HabitatHost() (habitatHost string, err error)
     PublicKey() (publicKey crypto.PublicKey, err error)
}
```
So that we can create a wrapper around and reuse this: https://github.com/bluesky-social/indigo/blob/bfc7f487e4ed5ece2c07a1bb309ba1d32dede33e/atproto/identity/identity.go#L116-L123